app/controllers/spree/retail/shopify/hooks/refunds_controller.rb

```diff
@@ -0,0 +1,24 @@
+module Spree
+  module Retail
+    module Shopify
+      module Hooks
+        class RefundsController < HooksController
+          def create
+            shopify_refund = ShopifyAPI::Refund.find(shopify_refund_id, params: { order_id: shopify_order_id })
+            Spree::Retail::Shopify::GenerateRefundOrder.new(shopify_refund).process
+
+            head :ok
+          end
+
+          def shopify_refund_id
+            params[:id]
+          end
+
+          def shopify_order_id
+            params[:order_id]
```

Comment:
Yeah, it seems impossible, since the route wouldn't match without a legit order ID. I guess I might be concerned the Shopify order couldn't be found?

In general I'm trying to ensure you've thought through the possible error cases and how this action ought respond in those cases.